### PR TITLE
rgw/notifications: add region to bucket ARN

### PR DESCRIPTION
**Pull request by yuvalif**
Fixes: https://tracker.ceph.com/issues/53186

Signed-off-by: Yuval Lifshitz <ylifshit@redhat.com>

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [x] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Review by anthonyeleven — APPROVED** [review]

**Comment by yuvalif** [comment]
teuthology passing: http://pulpito.front.sepia.ceph.com/yuvalif-2022-08-31_08:40:20-rgw:notifications-wip-yuval-fix-53186-distro-default-smithi/

**Comment by yuvalif** [comment]
jenkins test make check

**Comment by yuvalif** [comment]
@mattbenjamin can you please review?

File: src/rgw/rgw_notify.cc [hunk 1]
```diff
@@ -687,12 +687,15 @@ static inline void populate_event(reservation_t& res,
   // configurationId is filled from notification configuration
   event.bucket_name = res.bucket->get_name();
   event.bucket_ownerIdentity = res.bucket->get_owner() ? res.bucket->get_owner()->get_id().id : "";
-  event.bucket_arn = to_string(rgw::ARN(res.bucket->get_key()));
+  const auto region = res.store->get_zone()->get_zonegroup().get_api_name();
+  rgw::ARN bucket_arn(res.bucket->get_key());
+  bucket_arn.region = region;
```
Comment (mattbenjamin) [inline src/rgw/rgw_notify.cc:692]: ok, got it

**Review by mattbenjamin — APPROVED** [review]
lgtm

**Pull request merged**